compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2302,6 +2325,7 @@ nop_list_lift! {
     poly_existential_predicates; PolyExistentialPredicate<'a> => PolyExistentialPredicate<'tcx>
 }
 nop_list_lift! { bound_variable_kinds; ty::BoundVariableKind => ty::BoundVariableKind }
+nop_list_with_cached_type_info_lift! { clauses; Clause<'a> => Clause<'tcx> }
```

Comment:
naming question.. what is the meaning of `nop` here? This lift is definitely not a noop, as we do have check that the list has actually been interned by the `TyCtxt`

Reply:
That's what the nop means afaict; instead of re-interning the list, it's able to do the lifting by checking that the interned pointer originates from the tyctxt's arena. It's avoiding doing any actual work, hence "nop".